lib/Dashboard/RecommendationWidget.php

```diff
@@ -68,4 +88,23 @@ public function load(): void {
 		}
 		Util::addScript(Application::APP_ID, 'files_recommendation-dashboard');
 	}
+
+	public function getItems(string $userId, ?string $since = null, int $limit = 7): array {
+		$user = $this->userManager->get($userId);
+		if (!$user) {
+			return [];
+		}
+		$recommendations = $this->recommendationService->getRecommendations($user, $limit);
+
+		return array_map(function(IRecommendation $recommendation) {
+			$url = $this->urlGenerator->linkToRouteAbsolute('files.viewcontroller.showFile', ['fileid' => $recommendation->getNode()->getId()]);
+			$icon = $this->urlGenerator->linkToRouteAbsolute('core.Preview.getPreviewByFileId', [
```

Comment:
@tobiasKaminsky do you mean we should pre-check if a preview exists, and if yes, return the preview URL and if not return the mime type icon URL instead ?

Reply:
Yes, clients always need a real icon, may it be svg or png.

Reply:
At least in Qt/QML, this is not required as you can easily detect if the image loaded and add a fallback (e.g. mimetype icon). See e.g.  https://invent.kde.org/frameworks/kirigami/-/blob/master/src/controls/Avatar.qml#L235

Reply:
But we do not have any info about the mimetype.
And sending mimetype along side, and then have clients deteminate it and show correct placeholder is way more work than directly sending the correct image.
(widgets on Android / iOS are limited in opposite to our full apps, so we cannot do too much calculation/work there)